### ISSUE TITLE
Capability to extend and access some core classes in Plugins

### ISSUE THREAD

**Issue by bastiao** [bioinformatics-ua/dicoogle#128]
At least, these classes (or its content) should be available in Plugins:
- ServerSettings (access)
- DIMGeneric (capability to extend)
- Related classes: Patient-> Study -> Series  (capability to extend). 

Why?
- You are not able to check what DICOM nodes are registered, DICOM services default ports.  
- It is often common to play around DICOM Information Model, such as count number of Patients, Studies, Series and Images. The code is already available in core, but at the moment, the Plugins have two do a trick to extend it (or replicate the code). 

Approach: 
A possible approach is to study how we can accommodate it a dicoogle-commons library or (sdk-ext). Added them directly in SDK will make the core SDK complex IMHO.

**Comment by Enet4** [comment]
> ServerSettings (access)
>
> You are not able to check what DICOM nodes are registered, DICOM services default ports.

Read-only access to global settings is already available with #78. You may wish to see `ServerSettingsReader#getMoves()`.

**Comment by bastiao** [comment]
@Enet4 yes. But probably we need to extend to a writer too or any other way to write. 
But  last two points are also crucial. That's the main reason I opened the issue. To start thinking about it and come up with a good solution.

**Comment by Enet4** [comment]
So you also want write access... Sure, that changes things a bit.

**Comment by bastiao** [comment]
I think we can close this issue.
